common/flatpak-utils.c

```diff
@@ -5803,11 +5817,11 @@ add_extension (GKeyFile   *metakey,
   if (ref == NULL)
     return res;
 
-  files = flatpak_find_unmaintained_extension_dir_if_exists (extension, arch, branch, NULL);
```

Comment:
This part is a bit weird, because an unmaintained extension dir will not have a deploy file, nor a commit to look up in the repo. So, i think we could drop this and just make sure we correctly handle "upgrades" if OstreeRepo is NULL.

Reply:
Oops, yeah I didn't really understand what an unmaintained extension is when I wrote that but now I do (https://github.com/flatpak/flatpak/issues/167).

I think we can just avoid the call to `flatpak_load_deploy_data()` in `flatpak_extension_new()` in the unmaintained case, since the deploy data won't exist anyway.